job/server/src/main/java/alluxio/job/JobMasterContext.java

```diff
@@ -20,18 +22,41 @@
  */
 @ThreadSafe
 public final class JobMasterContext {
+  private final FileSystem mFileSystem;
+  private final FileSystemContext mFsContext;
   private final long mJobId;
   private final UfsManager mUfsManager;
 
   /**
+   * Creates a new instance of {@link JobMasterContext}.
+   *
+   * @param filesystem the Alluxio client used to contact the master
+   * @param fsContext the filesystem client's underlying context
    * @param jobId the job id
    * @param ufsManager the UFS manager
    */
-  public JobMasterContext(long jobId, UfsManager ufsManager) {
+  public JobMasterContext(FileSystem filesystem, FileSystemContext fsContext, long jobId,
```

Comment:
Why does the `JobMasterContext` have a `jobId`? Is this more of a `JobContext`?

Reply:
The of executing a job is:

- get a scheduled Job ID
- Select the executors for that job (`selectExecutors`)
- send the jobconfig to worker
- Worker executes the job based on the config (`runTask`)

the `selectExecutors` portion takes an argument `JobMasterContext` which provides the necessary information for the Job Master to determine which job worker to run the job on. I assume the original author called it JobMasterContext because the JobMaster is the only part that executes `selectExecutors` and the `runTask` only runs on the Job worker so it takes a `JobWorkerContext`